lib/model.js

```diff
@@ -1111,6 +1111,30 @@ class Model {
       }
     })
     .then(() => this.QueryInterface.createTable(this.getTableName(options), attributes, options, this))
+    .then(() => {
+      if(options.alter) {
+        return this.QueryInterface.describeTable(this.getTableName(options))
+          .then((columns) => {
+            var self = this;
+            var changes = []; // array of promises to run
+            _.each(attributes, function(columnDesc, columnName) {
+              if(!columns[columnName]) {
+                changes.push(() => self.QueryInterface.addColumn(self.getTableName(options), columnName, attributes[columnName]));
```

Comment:
These are needed because without the arrow function, the query calls immediately, but we need to execute them sequentially.